strawberryfields/gbs/subgraph.py

```diff
@@ -183,193 +178,98 @@ def random_search(
 describing the method, while the dictionary values are callable functions corresponding to the
 method."""
 
-RESIZE_DEFAULTS = {"method": "greedy-density"}
-"""dict[str, Any]: Dictionary to specify default parameters of options for resizing
-"""
-
-OPTIONS_DEFAULTS = {"heuristic": {"method": random_search}, "resize": RESIZE_DEFAULTS}
+OPTIONS_DEFAULTS = {"heuristic": {"method": random_search}}
 """dict[str, dict[str, Any]]: Options for dense subgraph identification heuristics. Composed of a
 dictionary of dictionaries with the first level specifying the option type, selected from keys
 ``"heuristic"`` and ``"resize"``, with the corresponding value a
 dictionary of options for that type.
 """
 
 
-def resize(
-        subgraphs: Iterable, graph: nx.Graph, target: int, resize_options: Optional[dict] = None
-) -> list:
-    """Resize subgraphs to a given size.
+def resize(subgraph: list, graph: nx.Graph, sizes: Optional[list] = None) -> list:
+    """Resize a subgraph to a range of input sizes.
 
-    This function resizes subgraphs to size ``target``. The resizing method can be fully
-    specified by the optional ``resize_options`` argument, which should be a dict containing any
-    of the following:
-
-    .. glossary::
+    This function uses a greedy approach to iteratively add or remove nodes one at a time to an
+    input subgraph to reach the range of sizes specified by ``sizes``.
 
-        key: ``"method"``, value: *str* or *callable*
-            Value can be either a string selecting from a range of available methods or a
-            customized callable function. Options include:
+    When growth is required, the algorithm picks individual nodes from the remainder of the graph
+    as candidates to add into the subgraph and picks the node with the highest degree relative to
+    the rest of the subgraph. This results in a graph that is one node larger, and if growth is
+    still required the algorithm performs the above procedure again.
 
-            - ``"greedy-density"``: resize subgraphs by iteratively adding/removing nodes that
-              result in the densest subgraph among all candidates (default)
-            - ``"greedy-degree"``: resize subgraphs by iteratively adding/removing nodes that
-              have the highest/lowest degree.
-            - *callable*: accepting ``(subgraphs: list, graph: nx.Graph,
-              target: int, resize_options: dict)`` as arguments and returning a ``list``
-              containing the resized subgraphs as lists of nodes, see :func:`greedy_density` for
-              an example signature
+    When shrinking is required, the algorithm picks individual nodes from within the subgraph to
+    remove, choosing the nodes with lowest degree relative to the subgraph. In both growth and
+    shrink phases, ties for addition/removal with nodes of equal degree are settled by uniform
+    random choice.
 
```

Comment:
Should we have Example usage? Also in other parts of the refactored subgraph and sample modules?